File: angular-frontend/src/app/models/employee.ts

//zelfde field namen als DTO backend!!
export class Employee{
  id?:number;
  firstName:string;
  lastName:string;
  email:string;
  rrn:string;
  phoneNumber:string;

  constructor(firstName: string, lastName: string, email: string,rrn: string,phoneNumber:string, id?: number,) {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.rrn = rrn;
    this.phoneNumber=phoneNumber;
  }
}

File: springboot-backend/src/main/java/com/example/springbootbackend/domain/Employee.java
package com.example.springbootbackend.domain;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;
import org.hibernate.validator.constraints.Length;


@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotEmpty(message = "firstname can not be empty")
    private String firstName;

    @NotEmpty(message = "lastname can not be empty")
    private String lastName;

    @Email(message = "Invalid email format")
    private String email;

    //rijksregisternummer
    @NotEmpty(message = "RRN can not be empty")
    @Length(min = 11, max = 11,message = "Length must be 11 numbers")
    private String rrn;

   @NotEmpty(message = "Phonenumber can not be empty")
    private String phoneNumber;

}
